src/components/demo-tabs.tsx

```diff
@@ -0,0 +1,183 @@
+import React from 'react';
+import { StaticQuery, graphql } from 'gatsby';
+import { ContentPageData } from '../types';
+import AuthorsList from './authors-list';
+import EditLink from './edit-link';
+import { ParsedContent } from '../util/ParsedContent';
+
+const query = graphql`
+  query DemoTabsContent {
+    content: markdownRemark(
+      frontmatter: {
+        title: { eq: "Installing via Package Manager" }
+        section: { eq: "Content" }
+      }
+    ) {
+      id
+      html
+      tableOfContents
+      frontmatter {
+        title
+        description
+      }
+      fields {
+        slug
+        authors
+      }
+      parent {
+        ... on File {
+          relativePath
+        }
+      }
+    }
+  }
+`;
+
+export interface Props {
+  parsedContent?: ParsedContent;
+}
+
+class DemoTabs extends React.Component<Props> {
+  public parsedContent: ParsedContent;
+
+  public parsedFragment: DocumentFragment | null;
+
+  public articleRef: React.RefObject<HTMLElement>;
+
+  public selectRef: React.RefObject<HTMLSelectElement>;
+
+  public optionMap: WeakMap<HTMLOptionElement, HTMLElement>;
+
+  public options: HTMLOptionElement[];
+
+  public sections: HTMLElement[];
+
+  public currentSection: HTMLElement | null;
+
+  public constructor(props: Props) {
+    super(props);
+    this.parsedContent = (props && props.parsedContent) || new ParsedContent();
+    this.articleRef = React.createRef();
+    this.selectRef = React.createRef();
+    this.optionMap = new WeakMap<HTMLOptionElement, HTMLElement>();
+    this.options = [];
+    this.sections = [];
+    this.currentSection = null;
+    this.parsedFragment = null;
+  }
+
+  public componentDidMount() {
+    this.populate();
+  }
+
+  public populate() {
+    if (this.parsedFragment === this.parsedContent.documentFragment) return;
+
+    /* eslint-disable */
+
+    const {
+      parsedContent,
+      parsedFragment,
+      articleRef: { current: articleNode },
+      selectRef: { current: selectNode },
+    } = this;
+
+    const splicedNodes = [
+      ...this.options.splice(0, this.options.length),
+      ...this.sections.splice(0, this.sections.length),
+    ];
+
+    this.currentSection = null;
+
+    // @ts-ignore
```

Comment:
Why is this needed?

Reply:
TypeScript's includes layered `lib.dom` definitions. In the old days when I wrote `.ts` I usually included them in the `tsconfig.json` but I did not want to make this potentially breaking project-wide change myself.

I've since moved to typed-`.js` for almost 2 years now and I am not sure if I included `/// <reference…` directives in a mixed project works.

This looks like this in a `.js` file in my other project(s):

```js
//@ts-check
/// <reference lib="dom" />
/// <reference lib="dom.iterable" />
/// <reference lib="es2020" />
```

Based on how we want to go about `lib.dom.iterable`, we might be better off landing a separate PR so people can pull it and avoid accidental breaks.